File: su18/problem283.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        var count = 0;
        for (var i = 0; i < nums.Length; i++) {
            if (nums[i] != 0) {
                var temp = nums[i];
                nums[i] = 0;
                nums[count] = temp;
                count++;
            }
        }
    }
}

File: su19/problem283.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        nextZero = len(nums)
        i = 0
        while i < len(nums):
            if nums[i] != 0 and i > nextZero:
                nums[nextZero] = nums[i]
                nums[i] = 0
                i, nextZero = nextZero, i
            elif nums[i] == 0 and i < nextZero:
                nextZero = i
                i += 1
            else:
                i += 1

        return
            
        